sdk/core/Azure.Core/src/DynamicData/DynamicData.Converters.cs

```diff
@@ -10,16 +10,31 @@ namespace Azure.Core.Dynamic
 {
     public partial class DynamicData
     {
-        // From: https://github.com/Azure/autorest.csharp/blob/d835b0b7bffae08c1037ccc5824e928eaac55b96/src/assets/Generator.Shared/TypeFormatters.cs#L14
-        private const string RoundtripZFormat = "yyyy-MM-ddTHH:mm:ss.fffffffZ";
+        internal const string RoundTripFormat = "o";
+        internal const string UnixFormat = "x";
 
-        private class Rfc3339DateTimeConverter : JsonConverter<DateTime>
+        internal class DynamicDateTimeConverter : JsonConverter<DateTime>
         {
+            public string Format { get; }
+
+            public DynamicDateTimeConverter(string format)
+            {
+                Format = format;
+            }
+
             public override DateTime Read(
                 ref Utf8JsonReader reader,
                 Type typeToConvert,
                 JsonSerializerOptions options)
             {
+                if (Format == UnixFormat)
```

Comment:
Unless we plan to reserve "X" as something distinct from "x", should we do a case insensitive compare here?

Reply:
Cool, good point.  Updated.